airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -399,4 +413,94 @@ private static long getEpoch(Record record, String fieldName) {
     return record.get(fieldName, LocalDateTime.class).toEpochSecond(ZoneOffset.UTC);
   }
 
+  @Override
+  public Map<String, Stream<JsonNode>> exportDatabase(final String schema) throws IOException {
+    final List<String> tables = listTables(schema);
+    final Map<String, Stream<JsonNode>> result = new HashMap<>();
+    for (final String table : tables) {
+      result.put(table, exportTable(schema, table));
+    }
+    return result;
+  }
+
+  /**
+   * List tables from @param schema and @return their names
+   */
+  private List<String> listTables(final String schema) throws IOException {
+    if (schema != null) {
+      return database.query(context -> context.meta().getSchemas(schema).stream()
+          .flatMap(s -> context.meta(s).getTables().stream())
+          .map(Named::getName)
+          .collect(Collectors.toList()));
+    } else {
+      return List.of();
+    }
+  }
+
+  private Stream<JsonNode> exportTable(final String schema, final String tableName) throws IOException {
+    final String tableSql = String.format("%s.%s", schema, tableName);
+    try (final Stream<Record> records = database.query(ctx -> ctx.select(DSL.asterisk()).from(tableSql).fetchStream())) {
```

Comment:
Is that how i am supposed to apply your suggestion from https://github.com/airbytehq/airbyte/pull/1619/#discussion_r556017334 ?

Reply:
yup. looks good.